torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -13,6 +13,34 @@
 )
 
 
+###############################################################################
+# Kernel Side Table
+
+# We cannot put Triton Kernels into the FX graph as the graph nodes
+# do not support arbitrary functions
+# Use a side table
+kernel_side_table = threading.local()
+kernel_side_table.table = []
+
+
+# Returns index on the table
+def add_kernel_to_table(kernel) -> int:
+    # TODO(oulgen): Dedup kernels by code id
```

Comment:
Deferring this to a follow up but I could not come up with an amortized O(1) way to dedup. Assuming code id is unique enough for the dedup, we would need a way to see if kernel is already in the table as well as fetch it based on an id. Off the top of my head, we need a dict data structure that has O(1) search on both the key and the value, does Python have something like this builtin or would we need to build that? should be quite simple

Reply:
Python has `set({})`

Reply:
Sorry, I see the problem. Would something like a dict[int, kernel] work? whenever you insert into the dict you use len(dict) to get the next valid key.

Reply:
How do I check whether the kernel is unique? I guess I can keep a set and a dict, right?